rest/src/test/scala/io/udash/rest/SomeServerApiImpl.scala

```diff
@@ -0,0 +1,9 @@
+package io.udash.rest
+
+import scala.concurrent.Future
+
+final class SomeServerApiImpl {
+  @GET
+  def thingy(param: Int): Future[String] = Future.successful((param - 1).toString)
+}
```

Comment:
Could you please test the more esoteric cases here? (abstract method, inherited method, `val`, `lazy val`)

Reply:
All public methods are scanned. It doesn't matter if they're `val`s, `lazy val`s, abstract or concrete. Methods can be excluded with `@ignore` annotation if desired. I'm already using this technique in Dolang a lot.

Reply:
I know, I just meant a more complex / integration example, not thorough testing of corner cases etc.

Reply:
I am very confident that this works, mainly because of Dolang usage. I will add some more tests in `scala-commons` for that, there's nothing Udash specific about this.

Reply:
I'm not going to die on that hill, but I'm a fan of testing the basic use-case (blackbox, integrational) a bit anyway even when using a lib - we have no guarantess (in Udash), that the tests you mentioned will always / still exist on an upgrade, so we could detect the biggest issues then. That might be just a personal paranoia, so feel free to merge.

Reply:
I added a test involving `@POST`, prefix method and a `val`